src/transformers/models/jamba/modeling_jamba.py

```diff
@@ -718,6 +719,34 @@ def update(
 
         return self.key_cache[layer_idx], self.value_cache[layer_idx]
 
+    def get_seq_length(self, layer_idx: Optional[int] = 0) -> int:
+        """Returns the sequence length of the cached states. A layer index can be optionally passed."""
+        if len(self.key_cache) <= layer_idx:
+            return 0
+        if self.layers_block_type[layer_idx] == "mamba":
+            raise ValueError("Can't return seq_length from Mamba layers cache as it doesn't have a sequence length dimension.")
+        return self.key_cache[layer_idx].shape[-2]
```

Comment:
would rather not have this, cache positions SHOULD be passed. That is here in llama for legacy